lib/ecto.ex

```diff
@@ -595,4 +595,21 @@ defmodule Ecto do
       true
     end
   end
+
+  @spec load(Ecto.Schema.t | map(), map() | Keyword.t | {list, list}) :: Ecto.Schema.t | map()
```

Comment:
btw, here's how it can be used end-to-end. Not sure it achieves Michal's original goal of making this stuff much easier, but perhaps it's a start:

```elixir
result = Ecto.Adapters.SQL.query!(HexWeb.Repo, "SELECT * FROM users", [])
columns = Enum.map(result.columns, &String.to_atom/1)
Enum.map(result.rows, fn row -> Ecto.load(HexWeb.User, Enum.zip(columns, row)) end)
#=> [%HexWeb.User{}, ...]
```